crates/iota-sdk-types/src/crypto/intent.rs

```diff
@@ -52,6 +77,47 @@ impl Intent {
     pub fn app_id(self) -> IntentAppId {
         self.app_id
     }
+
+    pub fn iota_app(scope: IntentScope) -> Self {
+        Self {
+            version: IntentVersion::V0,
+            scope,
+            app_id: IntentAppId::Iota,
+        }
+    }
+
+    pub const fn iota_transaction() -> Self {
+        Self {
+            scope: IntentScope::TransactionData,
+            version: IntentVersion::V0,
+            app_id: IntentAppId::Iota,
+        }
+    }
+
+    pub const fn personal_message() -> Self {
+        Self {
+            scope: IntentScope::PersonalMessage,
+            version: IntentVersion::V0,
+            app_id: IntentAppId::Iota,
+        }
+    }
+
+    pub const fn consensus_app(scope: IntentScope) -> Self {
+        Self {
+            scope,
+            version: IntentVersion::V0,
+            app_id: IntentAppId::Consensus,
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl FromStr for Intent {
```

Comment:
This used to allow with and without `0x` prefix, I'm not sure that's the case anymore.